sinks/httpsink_test.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2017 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sinks
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api"
+
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+func TestUpdateEvents(t *testing.T) {
+	stopCh := make(chan bool, 1)
+	doneCh := make(chan bool, 1)
+
+	got := bytes.NewBuffer(nil)
+	seenRequests := make([]*http.Request, 0)
+	mockStatus := http.StatusOK
+
+	// Make a test server to send stuff too... it just copies its input to the
+	// `got` buffer and records the request.
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		seenRequests = append(seenRequests, r)
+		io.Copy(got, r.Body)
+		w.WriteHeader(mockStatus)
+	}))
+	defer srv.Close()
+
+	testPod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			SelfLink:  "/api/version/pods/foo",
+			Name:      "foo",
+			Namespace: "baz",
+			UID:       "bar",
+		},
+		Spec: v1.PodSpec{},
+	}
+	podRef, err := v1.GetReference(api.Scheme, testPod)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	evt := makeFakeEvent(podRef, api.EventTypeWarning, "CreateInCluster", "Fake pod creation event")
+
+	// 1. Try with a synchronous channel
+	sink := NewHTTPSink(srv.URL, false, 0)
+	go func() {
+		sink.Run(stopCh)
+		doneCh <- true
+	}()
+
+	// Send the event
+	sink.UpdateEvents(evt, nil)
+	stopCh <- true
+	<-doneCh
+
+	if got.Len() == 0 {
+		t.Errorf("Sent logs but didn't read any back")
+	}
+
+	// 2. Try with the server returning 500's, test retries
+	got.Truncate(0)
+	seenRequests = make([]*http.Request, 0)
+	sink = NewHTTPSink(srv.URL, false, 10)
+
+	go func() {
+		sink.Run(stopCh)
+		doneCh <- true
+	}()
+
+	// Send the event, sleep to ensure the request is attempted
+	mockStatus = http.StatusInternalServerError
+	sink.UpdateEvents(evt, nil)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
We should probably have a standard like 

// TODO: SLEEP 

For trace debugging in case we hit flakes.